integration-tests/chain-signatures/build.rs

```diff
@@ -80,11 +80,22 @@ async fn build_multichain_contract(release: bool) -> anyhow::Result<ExitStatus>
     .await
 }
 
+fn rerun_directory<T: AsRef<Path> + ?Sized>(dir: &T) {
+    println!("cargo:rerun-if-changed={}", dir.as_ref().to_str().unwrap());
+
+    for entry in fs::read_dir(dir).unwrap() {
+        let entry = entry.expect("cannot access file in src directory");
+        let path = entry.path();
+        if path.is_dir() {
+            // only look at directories for timestamps
+            rerun_directory(&path);
+        }
+    }
+}
+
 fn main() -> anyhow::Result<()> {
     println!("cargo:rerun-if-changed=build.rs");
-    println!("cargo:rerun-if-changed=../../contract/");
-    println!("cargo:rerun-if-changed=../../keys/");
-    println!("cargo:rerun-if-changed=../../node/");
+    rerun_directory("../../chain-signatures/");
```

Comment:
will this result multiple rerun if more than one dir changes?

Reply:
This build script only gets ran when you build the integration-tests package, so it'll only be ran once. The `rerun_directory` just has it listens to specific timestamps for a list of directories, and if any of them changes, it'll trigger a rebuild the next time we go to build this test package